action_plugins/load_profile/__init__.py

```diff
@@ -0,0 +1,165 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from __future__ import annotations
+
+import enum
+import logging
+import threading
+import time
+from typing import List, TYPE_CHECKING
+from xml.etree import ElementTree
+
+from PySide6 import QtCore
+from PySide6.QtCore import Property, Signal, Slot
+
+from gremlin import error, event_handler, input_devices, \
+    joystick_handling, mode_manager, shared_state, util
+from gremlin.base_classes import AbstractActionData, AbstractFunctor, \
+    DataCreationMode, Value
+from gremlin.config import Configuration
+from gremlin.profile import Library
+from gremlin.types import ActionProperty, AxisMode, InputType, PropertyType
+
+from gremlin.ui.action_model import SequenceIndex, ActionModel
+
+if TYPE_CHECKING:
+    from gremlin.ui.profile import InputItemBindingModel
+
+
+class LoadProfileFunctor(AbstractFunctor):
+
+    """Executes a load profile action callback."""
+
+    def __init__(self, action: LoadProfileData):
+        super().__init__(action)
+
+        self._profile_filename = action.profile_filename
+
+    def __call__(
+        self,
+        event: event_handler.Event,
+        value: Value
+    ) -> None:
+
+        if not self._should_execute(value):
+                return
+
+        logging.getLogger("system").debug(
+            f"Load profile ...{self._profile_filename}"
+        )
+
+class LoadProfileModel(ActionModel):
+
+    fileChanged = Signal()
+
+    def __init__(
+            self,
+            data: AbstractActionData,
+            binding_model: InputItemBindingModel,
+            action_index: SequenceIndex,
+            parent_index: SequenceIndex,
+            parent: QtCore.QObject
+    ):
+        super().__init__(data, binding_model, action_index, parent_index, parent)
+
+    def _qml_path_impl(self) -> str:
+        return "file:///" + QtCore.QFile(
+            "core_plugins:load_profile/LoadProfileAction.qml"
+        ).fileName()
+
+    def _icon_string_impl(self) -> str:
+        return LoadProfileData.icon
+
+    def _get_profile_filename(self) -> str:
+        return self._data.profile_filename
+
+    def _set_profile_filename(self, value: str) -> None:
+        if str(value) == self._data.profile_filename:
+            return
+        self._data.profile_filename = str(value)
+        self.fileChanged.emit()
+
+    profile_filename = Property(
+        str,
+        fget=_get_profile_filename,
+        fset=_set_profile_filename,
+        notify=fileChanged
+    )
+
+
+class LoadProfileData(AbstractActionData):
+
+    """Model of a load profile action."""
+
+    version = 1
+    name = "Load Profile"
+    tag = "load-profile"
+    icon = "\uF3B9"
+
+    functor = LoadProfileFunctor
+    model = LoadProfileModel
+
+    properties = [
```

Comment:
You probably don't want this, as that would trigger a profile load when the input is pressed and again when released. The better one is `ActivateOnPress` as that only is the default behavior not the one that it is restricted to. You also may want to add the `AlwaysExecute` property as otherwise you won't be able to load a profile when the current profile is paused. In R14 that "run when paused" is a per-action-class setting rather than per-instance as was the case in R13.

Reply:
Noted.